src/app/clients/clients.datasource.ts

```diff
@@ -34,25 +33,43 @@ export class ClientsDataSource implements DataSource<any> {
    * @param {number} pageIndex Page number.
    * @param {number} limit Number of clients within the page.
    */
-  getClients(orderBy: string = '', sortOrder: string = '', pageIndex: number = 0, limit: number = 10, showClosedAccounts: boolean = true) {
+  getClients(
+    orderBy: string = "",
+    sortOrder: string = "",
+    pageIndex: number = 0,
+    limit: number = 10,
+    showClosedAccounts: boolean = true
+  ) {
     this.clientsSubject.next([]);
-    this.clientsService.getClients('id', 'ASC', pageIndex * limit, limit)
-      .subscribe((clients: any) => {
-        if (showClosedAccounts) {
-          clients.pageItems = clients.pageItems;
-        } else {
-          clients.pageItems = clients.pageItems.filter((client: any) => client.status.value !== 'Closed');
+    let countryId = JSON.parse(sessionStorage.getItem("selectedCountry"))?.id;
+    if (countryId) {
```

Comment:
The subscribe can be centralized since it is the same for both the if and the else. Change this block to

```
let Observable<any> clientsObs ;
if (countryId) {
     clientsObs = this.clientsService
        .getClientsByCountry("id", "ASC", pageIndex * limit, limit, countryId);
    } else {
      clientsObs =this.clientsService.getClients("id", "ASC", pageIndex * limit, limit)
   }
   clientsObs.subscribe((clients: any) => {
        if (!showClosedAccounts) {

          clients.pageItems = clients.pageItems.filter((client: any) => client.status.value !== "Closed");
        }
        this.recordsSubject.next(clients.totalFilteredRecords);
        this.clientsSubject.next(clients.pageItems);
      });
    

```